packages/payload/src/utilities/flattenTopLevelFields.ts

```diff
@@ -18,38 +22,153 @@ import {
 } from '../fields/config/types.js'
 
 type FlattenedField<TField> = TField extends ClientField
-  ? FieldAffectingDataClient | FieldPresentationalOnlyClient
-  : FieldAffectingData | FieldPresentationalOnly
+  ? { accessor?: string; labelWithPrefix?: string } & (
+      | FieldAffectingDataClient
+      | FieldPresentationalOnlyClient
+    )
+  : { accessor?: string; labelWithPrefix?: string } & (FieldAffectingData | FieldPresentationalOnly)
 
 type TabType<TField> = TField extends ClientField ? ClientTab : Tab
 
 /**
- * Flattens a collection's fields into a single array of fields, as long
- * as the fields do not affect data.
+ * Options to control how fields are flattened.
+ */
+type FlattenFieldsOptions = {
+  /**
+   * If true, nested fields inside `group` fields will be lifted to the top level
+   * and given contextual `accessor` and `labelWithPrefix` values.
+   * Default: false.
+   */
+  extractFieldsToTopFromGroupFields?: boolean
```

Comment:
I'd make this arg name more generic and include tabs with names.

`moveSubFieldsToTop` or some other name.